### PR TITLE
Weights

### PR DESCRIPTION
**Pull request by mebrooks**
I added a test that makes sure a model with aggregated data and weights is identical to its non-aggregated version. Tests of binomial models that had proportions as the response were switched to the `cbind(success, failure)` specification even though the proportion version still works.

### CODE REVIEW

**Comment by mebrooks** [comment]
More tests may need to be added in the future (e.g., models with zero-inflation and/or dispersion components).

File: glmmTMB/tests/testthat/test-families.R [hunk 1]
```diff
@@ -33,13 +33,6 @@ test_that("binomial", {
     expect_equal( logLik(mod4)     , logLik(mod5) )
     expect_equal( fixef(mod4)$cond , fixef(mod5)$cond )
 
-    ## Now with extra weights
-    dd$w <- 2
-    mod6 <- glmmTMB(cbind(success,failure)~1,family=binomial,data=dd,weights=w)
-    mod7 <- glmmTMB(prop~1,weights=size*w,family=binomial,data=dd)
```
Comment (kaskr) [inline glmmTMB/tests/testthat/test-families.R]: @mebrooks  Could we keep this test ? We still allow prop/weights right ?

File: glmmTMB/R/glmmTMB.R [hunk 2]
```diff
@@ -78,12 +80,36 @@ mkTMBStruc <- function(formula, ziformula, dispformula,
...
+        size <- rep(1, nobs)
+      } else { #proportions
+          yobs <- weights * yobs
+          size <- weights
+          weights <- rep(1, nobs)
+        }
+      }
+    }
   }
+  if (is.null(size)) size <- rep(1, nobs)
```
Comment (kaskr) [inline glmmTMB/R/glmmTMB.R]: Better to pass `numeric(0)` as default `size`. That'll ensure it's not used somewhere by accident.

File: glmmTMB/tests/testthat/test-families.R (hunk 1 above)
Comment (kaskr) [reply glmmTMB/tests/testthat/test-families.R]: In that case I'd suggest modifying the test to check the likelihood of each of the two fits against `glm`.

Comment (mebrooks) [reply glmmTMB/tests/testthat/test-families.R]: @kaskr I put back the check of the `fixef` but removed the `logLik` check.

Comment (mebrooks) [reply glmmTMB/tests/testthat/test-families.R]: @kaskr Sorry, I didn't see the suggestion to compare to `glm`. I just made *another* commit that does this.

**Pull request merged**